compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2231,6 +2231,10 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         // ambiguous impls. The latter *ought* to be a
         // coherence violation, so we don't report it here.
 
+        if self.tcx.opt_rpitit_info(obligation.cause.body_id.to_def_id()).is_some() {
```

Comment:
It's not generally sound to skip reporting an error here, without at least delaying a bug or something else. Can you explain why you're doing this?